doc/developer/feature-branch.rst

```diff
@@ -0,0 +1,109 @@
+Specific Feature Development Branch
+===================================
+
+This document describes the process to request and manage a dedicated branch
+on the FRR GitHub repo for the development of new features.
+
+Why dedicated branch to develop new feature?
+--------------------------------------------
+
+Some new features need to be developed in a collaborative way
+e.g. IS-IS Segment Routing, BGP-LS, PCC … and this specific development will
+take time.
+
+Providing a common Git reference will ease cooperation between partners.
+Indeed, it is hard to synchronise independent Git Repo. Only main FRR GitHub
+repo are known by everybody and private forks to create Pull Request are not
+known by everybody. In addition, there no space on FRR web site or wiki to
+publicly announce that someone is working on a given feature. FRR community
+is, in general, aware about the development of a feature when Pull Request is
+published, which is be too late for good cooperation.
+
+Feature Branch characteristics
+------------------------------
+
+A dedicated branch for a specific feature collaborative development is created
+as follow in the GitHub FRR repo (http://github.com/FRRouting/frr):
+
+- dev-XXX where XXX is the acronyme of the feature e.g. dev-bgpls
+- Feature branch is not linked to CI/CD chain, so will not trigger any tests
+  not compilation when a Pull Request is submitted to this branch
+- The README.md file from master branch is modified as follow:
+
+::
+
+  FRRouting XXX Development Branch
+  ================================
+
+  This special FRRouting branch is dedicated to the development of XXX.
+  The code located under this branch is a **Work In Progress**. The code may
+  compile or not, may run or not. In any case, this code **MUST NOT BE USED IN
+  PRODUCTION**.
+
+  FRRouting
+  ---------
+
+  FRR is free software that implements a ...
+
+
+Process to Request Feature Branch
+---------------------------------
+
+Only feature listed in FRR Wiki Features Requests
+(see https://github.com/FRRouting/frr/wiki/Feature-Requests) can be subject of
+this process.
+
+Everybody *(NOTE: another option is to limit this possibility to the mainteners
+only which will simplify the process, but close the door to not mainteners)*
+could ask for a new feature branch by following the bellow process:
+
+- **Prerequisit:** If feature is not listed in FRR Wiki Features Requests,
+  first create a new issue in FRR GitHub to request the new feature and wait
+  the feature is added to the Wiki
+- **Request:** [Send a simple mail is send to @frr-dev mailing list | Fill a
+  new issue | Update issue of the feature] *(NOTE: we need to choose the
+  appropriate method)* with a short description of the devlopment with the
+  issue number *(if mail method is choosen)*, and, if known, the list of
+  partners that express to work on. *(NOTE: these are the minimum information
+  I thought of. Of course additional information could be added)*
```

Comment:
this should not prevent future partners to collaborate subsequently in the dev branch